src/components/TimeTracker.tsx

```diff
@@ -1,8 +1,180 @@
+    isDurationLimitExceeded, 
+    convertDurationToHrsMinsSecs, 
+    calculateEndTime,
+} from "../utils/hoursAndMinutes";
+import { groupEntriesByWeek } from "../utils/groupEntriesByWeek";
+import { RootState, AppDispatch } from "../redux/store";
+import { FocusEvent, KeyboardEvent, Type } from "../types/types";
+import { checkString } from "../utils/checkString";
+import { createTimeEntry } from "../redux/clockifyThunk";
+import { TimeEntries } from "./TimeEntries";
 
 export function TimeTracker(){
+    const { isModalOpen, currentTask, selectedProject, selectedClient, data} = useSelector((state: RootState) => state.clockify)
+    const {startTime, endTime, duration, taskName} = currentTask
+
+    const timeStart = new Date(startTime)
+    const timeEnd = new Date(endTime)
+    
+    const [showProjects, setShowProjects] = useState(false)
+    const [startDateTime, setStartDateTime] = useState(getFormattedTime(timeStart))
+    const [endDateTime, setEndDateTime] = useState(getFormattedTime(timeEnd))
+    const [totalDuration, setDuration] = useState(duration)
+
+    const dispatch = useDispatch<AppDispatch>()
+
+    const handleStartTimeBlur = (type: Type, e: FocusEvent): void => {
+        const {isValid, hours, minutes} = convertToHoursAndMinutes(e.target.value)
+        const newStart = new Date(timeStart)
+        newStart.setHours(hours, minutes)
+        if (!isValid || isDurationLimitExceeded(newStart, timeEnd)) {
+            type === 'add' && setStartDateTime(getFormattedTime(timeStart))
+            return
+        }
+        if(type === 'add'){
+            dispatch(updateStartTime(newStart.toString()))
+            setStartDateTime(getFormattedTime(newStart))
+        }
+        if(hours > timeEnd.getHours() || minutes > timeEnd.getMinutes()){
+            type === 'add' && timeEnd.setDate(timeEnd.getDate() + 1)
+        }
+        
+        const { hrs, mins } = calculateTimeDifference(new Date(newStart), timeEnd)
+        
+        type === 'add' && dispatch(updateEndTime(timeEnd.toString()))
+
+        if(hrs <= 999){
+            type === 'add' && setDuration(formatTime(hrs, mins, 0))
+        }
+    }
+
+    const handleEndTimeBlur = (type: Type, e: FocusEvent): void => {
+        const {isValid, hours, minutes} = convertToHoursAndMinutes(e.target.value)
+        const newEnd = new Date(timeEnd)
+        newEnd.setHours(hours, minutes)
+        if(!isValid || isDurationLimitExceeded(timeStart, newEnd)) {
+            type === 'add' && setEndDateTime(getFormattedTime(timeEnd))
+            return
+        }
+
+        if(timeStart.getHours() > hours || timeStart.getMinutes() > minutes){
+            newEnd.setDate(newEnd.getDate() + 1)
+        }
+        const { hrs, mins } = calculateTimeDifference(new Date(newEnd), timeStart)
+
+        if(type === 'add'){
+            setEndDateTime(getFormattedTime(newEnd))
+            dispatch(updateEndTime(newEnd.toString()))
+        }
+
+        if(hrs <= 999){
```

Comment:
Search for Magic Numbers from the Clean code perspective and see if there's something you can do about this here.

Reply:
https://github.com/PreethiKalyani99/Clockify-Typescript/commit/5191e28172351d61ca7314d8442318a6b37f5c0f

Used enum for constants